### PR TITLE
[SPARK-38743][SQL][TEST] Test the error class: MISSING_STATIC_PARTITION_COLUMN

### PR DESCRIPTION
**Pull request by rangareddy**
### What changes were proposed in this pull request?

This PR aims to add a test for the error class **MISSING_STATIC_PARTITION_COLUMN** to **QueryCompilationErrorsSuite**.

### Why are the changes needed?

The changes improve test coverage, and document expected error messages in tests.

### Does this PR introduce _any_ user-facing change?

No, test-only.

### How was this patch tested?

By running new test:

```sh
$ build/sbt "sql/testOnly *QueryCompilationErrorsSuite*"
```

### CODE REVIEW

**Comment by rangareddy** [comment]
Hi @maxgekk

Could you please review this PR?

**Comment by rangareddy** [comment]
Hi @MaxGekk 

Could you please review the PR?

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 1]
```diff
@@ -679,6 +679,32 @@ class QueryCompilationErrorsSuite
...
+           |USING parquet
+           |PARTITIONED BY (c)
+           |""".stripMargin)
+
+      val df = Seq(("A", 123, "C")).toDF("a", "b", "c")
+      checkError(
+        exception = intercept[AnalysisException] {
+          df.write.mode("append").saveAsTable(s"$tableName")
+        },
+        errorClass = "_LEGACY_ERROR_TEMP_1163",
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:698]: How does this match to `MISSING_STATIC_PARTITION_COLUMN`?

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 2]
```diff
@@ -679,6 +679,32 @@ class QueryCompilationErrorsSuite
       context = ExpectedContext("", "", 7, 13, "CAST(1)")
     )
   }
+
+  test("SPARK-38743 - Test the error class: MISSING_STATIC_PARTITION_COLUMN") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:683]: Does `MISSING_STATIC_PARTITION_COLUMN` exist?

**Pull request closed without merging**